service-worker.js

```diff
@@ -15,4 +15,8 @@ var preCachedRessources = [
 
 toolbox.precache(preCachedRessources);
 
-toolbox.router.get('/(.*)', toolbox.cacheFirst, swOptions);
+toolbox.router.get('/*', toolbox.cacheFirst, swOptions);
+toolbox.router.get('/*', toolbox.cacheFirst, { origin: "cdnjs.cloudflare.com" });
+toolbox.router.get('/*', toolbox.cacheFirst, { origin: "cdn.jsdelivr.net" });
+toolbox.router.get('/*', toolbox.cacheFirst, { origin: "unpkg.com" }); // for repl
+toolbox.router.post('/*', toolbox.cacheFirst, { origin: "algolia.net" }); // Cache Algolia search response
```

Comment:
I wanted to keep search even once offline. The goal here is to cache some responses from Algolia because it does not support offline searching. I'm not sure about this since the user will be able to find what he already searched online.